frappe/public/js/frappe/list/base_list.js

```diff
@@ -764,10 +764,7 @@ class FilterArea {
 
 		const doctype_fields = this.list_view.meta.fields;
 		const title_field = this.list_view.meta.title_field;
-		const has_existing_filters = (
-			this.list_view.filters
-			&& this.list_view.filters.length > 0
-		);
+		const has_existing_filters = Array.isArray(this.list_view.filters);
```

Comment:
`isArray` check not strictly required. If it's truthy, it should be an Array.

Reply:
Yes, I realized it later, this is never undefined. Hence:

> This PR in the current state essentially removes the behavior where user defaults are auto-populated to link field filters.

Reply:
I'll remove all the code related to it once this is final.